src/edu/csus/ecs/pc2/core/model/Problem.java

```diff
@@ -2085,6 +2098,28 @@ public void setMemoryLimitMB(int memLimitInMB) {
         }
     }
 
+    /**
+     * Update sandbox command line program names based on whether the problem
+     * is using the internal sandbox and if it's interactive
+     * This routine makes sure that if we're using the internal sandbox, that the sandbox command line
+     * and sandbox program (script name) are correct.
+     */
+    private void updateSandboxInfo()
+    {
+        // we only set the sandbox program and command line if we're using the internal sandbox
+        // or, we haven't set anything yet.  We do the latter to prevent NPE's for someone requesting
+        // the sandbox name or command line even if the sandbox strings have not been defined yet
+        if(this.getSandboxType() == SandboxType.PC2_INTERNAL_SANDBOX || sandboxProgramName == null) {
+            if(isInteractive()) {
+                sandboxProgramName = Constants.PC2_INTERNAL_SANDBOX_INTERACTIVE_NAME;
+                sandboxCmdLine = Constants.PC2_INTERNAL_SANDBOX_INTERACTIVE_COMMAND_LINE;
+            } else {
+                sandboxProgramName = Constants.PC2_INTERNAL_SANDBOX_PROGRAM_NAME;                
+                sandboxCmdLine = Constants.PC2_INTERNAL_SANDBOX_COMMAND_LINE;
+            }
+        }
```

Comment:
The above bracket marks the end of `if (this.getSandboxType() == SandboxType.PC2_INTERNAL_SANDBOX || sandboxProgramName == null)`.  It seems like there should be an `else` clause to deal with what happens if the `if` clause is false.  (It may be that the only logical thing to do for the `else` at this point in time is something like throw an exception -- maybe an `ILLEGAL_STATE` exception -- but it _does_ seem like the code _should_ do some such thing if such a condition is arrived at.)

Reply:
The method is private.  It is used only by  other sandbox routines in the same module.  It does not matter if it "works" or not.  It just updates those 2 member variables is necessary.